randovania/game_description/game_description.py

```diff
@@ -213,14 +221,19 @@ def helper() -> Iterator[ResourceInfo]:
         for requirement_list in satisfiable_requirements:
             # If it's not satisfied, there's at least one IndividualRequirement in it that can be collected
             if not requirement_list.satisfied(resources, energy, database):
+                current_energy = energy
                 for individual in requirement_list.values():
                     # Ignore those with the `negate` flag. We can't "uncollect" a resource to satisfy these.
                     # Finally, if it's not satisfied then we're interested in collecting it
-                    if not individual.negate and not individual.satisfied(resources, energy, database):
+                    if not individual.negate and not individual.satisfied(resources, current_energy, database):
                         if individual.is_damage:
                             assert isinstance(individual.resource, SimpleResourceInfo)
                             yield from _resources_for_damage(individual.resource, database, resources)
                         else:
                             yield individual.resource
+                    elif individual.is_damage and individual.satisfied(resources, current_energy, database):
```

Comment:
Do you think you could make a test case for this function? It'll be a bit tricker as this function doesn't have tests yet though :(